spring-integration-ip/src/main/java/org/springframework/integration/ip/dsl/TcpInboundChannelAdapterSpec.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.dsl;
+
+import org.springframework.integration.dsl.MessageProducerSpec;
+import org.springframework.integration.ip.tcp.TcpReceivingChannelAdapter;
+import org.springframework.integration.ip.tcp.connection.AbstractConnectionFactory;
+import org.springframework.scheduling.TaskScheduler;
+
+/**
+ * A {@link MessageProducerSpec} for {@link TcpReceivingChannelAdapter}s.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class TcpInboundChannelAdapterSpec<S extends TcpInboundChannelAdapterSpec<S>>
```

Comment:
Well, I see that this generic variant works well in the target code.
So, now I'm not sure if we should restrict with target type without generic modification or leave it with that and let to extend in the future easy...
But, eh., typically we don't do that.
And I'm not sure that we are going to extend `TcpReceivingChannelAdapter` class to have some `TcpInboundChannelAdapterSpec` extension as well.